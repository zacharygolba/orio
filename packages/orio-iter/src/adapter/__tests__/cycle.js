// @flow

import Cycle from '../cycle'
import * as orio from '../../'

let subj

const cycle = producer => {
  const adapter = new Cycle(producer)

  jest.spyOn(adapter.producer, 'drop')
  return adapter
}

beforeEach(() => {
  const { producer } = orio.of(1, 2, 3)
  subj = cycle(producer)
})

afterEach(() => {
  jest.resetAllMocks()
})

test('#drop()', () => {
  subj.drop()

  expect(subj).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.producer.drop).toHaveBeenCalled()
})

test('#next()', () => {
  // First cycle...
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()

  // Second cycle...
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()

  // Third cycle...
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
})
