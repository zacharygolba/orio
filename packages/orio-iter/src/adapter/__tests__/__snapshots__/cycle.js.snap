// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#drop() 1`] = `
Cycle {
  "done": true,
  "index": 0,
  "items": Array [],
  "producer": Indexed {
    "drop": [Function],
    "index": 0,
    "source": Array [],
  },
  "useProducer": true,
}
`;

exports[`#drop() 2`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`#next() 1`] = `
Object {
  "done": false,
  "value": 1,
}
`;

exports[`#next() 2`] = `
Object {
  "done": false,
  "value": 2,
}
`;

exports[`#next() 3`] = `
Object {
  "done": false,
  "value": 3,
}
`;

exports[`#next() 4`] = `
Object {
  "done": false,
  "value": 1,
}
`;

exports[`#next() 5`] = `
Object {
  "done": false,
  "value": 2,
}
`;

exports[`#next() 6`] = `
Object {
  "done": false,
  "value": 3,
}
`;

exports[`#next() 7`] = `
Object {
  "done": false,
  "value": 1,
}
`;

exports[`#next() 8`] = `
Object {
  "done": false,
  "value": 2,
}
`;

exports[`#next() 9`] = `
Object {
  "done": false,
  "value": 3,
}
`;
